### PR TITLE
Concrete refactoring: step 1

### PR DESCRIPTION
**Pull request by ice1000**
This PR cleans up the parser.

### CODE REVIEW

File: cli/src/main/java/org/aya/cli/parse/error/DuplicatedModifierWarn.java [hunk 1]
```diff
@@ -16,7 +16,7 @@ public record DuplicatedModifierWarn(
 
   @Override
   public @NotNull Doc describe(@NotNull PrettierOptions options) {
-    return Doc.english("This modifier is duplicated with the previous, ignored.");
+    return Doc.english("This modifier is duplicated with the other, ignored.");
```
Comment (ice1000) [inline cli/src/main/java/org/aya/cli/parse/error/DuplicatedModifierWarn.java:19]:
```suggestion
    return Doc.english("This modifier is redundant, ignored.");
```

File: cli/src/main/java/org/aya/cli/parse/ModifierParser.java [hunk 2]
```diff
@@ -53,19 +81,45 @@ public ModifierParser(@NotNull Reporter reporter) {
...
+    for (var modi : modifiers) {
+      for (var implies : modi.data().implies) {
+        map.putIfAbsent(implies, modi.sourcePos());
       }
+    }
+
+    return Map.from(map).view()
+      .map((modi, pos) -> new WithPos<>(pos, modi)).toImmutableSeq();
+  }
+
```
Comment (ice1000) [inline cli/src/main/java/org/aya/cli/parse/ModifierParser.java:99]: https://stackoverflow.com/a/23701589/7083401

**Review by ice1000 — COMMENTED** [review]
@HoshinoTented

File: cli/src/main/java/org/aya/cli/parse/error/DuplicatedModifierWarn.java (hunk 1 above)
Comment (HoshinoTented) [reply cli/src/main/java/org/aya/cli/parse/error/DuplicatedModifierWarn.java:19]: 🤩

**Pull request merged**